### PR TITLE
update a lots of dependency to be compatible to actix_web 2.0

### PR DESCRIPTION
**Pull request by gzp-crey**
partially fixes: #214 

This is not a complete PR just a work in progress.

After your great PR of the tableentity i've tried to put together all the pieces of my project and it turned out for actix_rt (and actix_web) I need to bump the versions for some packages.
Unfortunately some of them had larger breaking change (ex http using self instead of &mut self in builders)

Some packages are still using the git version thus no crate publish is possible at the moment.

### CODE REVIEW

**Comment by gzp-crey** [comment]
fixing azure_sdk_auth_aad is blocked by:
  oauth2 and its reqwest dependecy. To continue I have to wait for a newer reqwest and oauth2 to be released.

fixing auzre_sdk_cosmos:
  As I see from issues it's being refactored thus I did not checked it. The change in the request builder might be something to consider during this refactor as the DocumentRequestExt pattern is not compatible to the "moved" Builder.

**Comment by sd2k** [comment]
@gzp-crey A new reqwest dependency was released a few days ago in case you have time to continue this! :smiley:

**Comment by gzp-crey** [comment]
Thanks, I'm checking it from time to time. Unfortunatelly oauth2's not been updated yet.

**Comment by gzp-crey** [comment]
After merging to main, I've managed to update the cosmos crate as well.
So only aad is missing due to the oauth2: https://github.com/ramosbugs/oauth2-rs/issues/79

**Comment by gzp-crey** [comment]
I think the PR is read to be merged back to main.

File: azure_sdk_core/src/modify_conditions.rs [hunk 1]
```diff
@@ -10,11 +10,15 @@ pub enum IfSinceCondition {
 }
 
 impl IfSinceCondition {
-    pub(crate) fn add_header(&self, builder: &mut Builder) {
+    pub(crate) fn add_header(&self, builder: Builder) -> Builder {
```
Comment (MindFlavor) [inline azure_sdk_core/src/modify_conditions.rs]: We should make sure to specify the `#[must_use]` directive (we can do this in a separate PR).

**Comment by MindFlavor** [comment]
This PR is awesome! 
Merging immediately!
